x-pack/test/functional/apps/visualize/precalculated_histogram.ts

```diff
@@ -24,37 +24,64 @@ export default function({ getService, getPageObjects }: FtrProviderContext) {
       return esArchiver.unload('pre_calculated_histogram');
     });
 
-    const initHistogramBarChart = async () => {
-      await PageObjects.visualize.navigateToNewVisualization();
-      await PageObjects.visualize.clickVerticalBarChart();
-      await PageObjects.visualize.clickNewSearch('histogram-test');
-      await PageObjects.visChart.waitForVisualization();
-    };
-
-    const getFieldOptionsForAggregation = async (aggregation: string): Promise<string[]> => {
-      await PageObjects.visEditor.clickBucket('Y-axis', 'metrics');
-      await PageObjects.visEditor.selectAggregation(aggregation, 'metrics');
-      const fieldValues = await PageObjects.visEditor.getField();
-      return fieldValues;
-    };
-
     it('appears correctly in discover', async function() {
       await PageObjects.common.navigateToApp('discover');
       const rowData = await PageObjects.discover.getDocTableIndex(1);
       expect(rowData.includes('"values": [ 0.3, 1, 3, 4.2, 4.8 ]')).to.be.ok();
     });
 
-    it('appears in the field options of a Percentiles aggregation', async function() {
-      await initHistogramBarChart();
-      const fieldValues: string[] = await getFieldOptionsForAggregation('Percentiles');
-      log.debug('Percentiles Fields = ' + fieldValues);
-      expect(fieldValues[0]).to.be('histogram-content');
-    });
+    describe('works in visualizations', () => {
+      before(async () => {
+        await PageObjects.visualize.navigateToNewVisualization();
+        await PageObjects.visualize.clickDataTable();
+        await PageObjects.visualize.clickNewSearch('histogram-test');
+        await PageObjects.visChart.waitForVisualization();
+        await PageObjects.visEditor.clickMetricEditor();
+      });
+
+      const renderTableForAggregation = async (
+        aggregation: string
+      ): Promise<string[][] | string[][][]> => {
+        await PageObjects.visEditor.selectAggregation(aggregation, 'metrics');
+        await PageObjects.visEditor.selectField('histogram-content', 'metrics');
+        await PageObjects.visEditor.clickGo();
+
+        return await PageObjects.visChart.getTableVisContent();
+      };
+
+      it('with percentiles aggregation', async () => {
+        const data = (await renderTableForAggregation('Percentiles')) as string[][];
+        expect(data[0]).to.have.property('length', 7);
+        // Percentile values are not deterministic, so we can't check for the exact values here,
+        // but just check they are all within the given range
+        // see https://github.com/elastic/elasticsearch/issues/49225
+        expect(data[0].every((p: string) => Number(p) >= 0.3 && Number(p) <= 5)).to.be(true);
+      });
+
+      it('with percentile ranks aggregation', async () => {
+        const data = await renderTableForAggregation('Percentile Ranks');
+        expect(data).to.eql([['0%']]);
```

Comment:
Nit: This test would be more helpful if it would test a different percentile rank than the default `0` - `0%` is a typical error value.

Reply:
Going to ignore this for now, since the the Percentile Ranks has the same non-deterministic problem than the others, so will rather fix that as soon as that's solved.